src/BlockTemplatesController.php

```diff
@@ -855,12 +855,36 @@ public function update_template_permalink( $value, $option ) {
 	 * @return string THe actual permalink assigned to the page. May differ from $permalink if it was already taken.
 	 */
 	protected function sync_endpoint_with_page( $page, $page_slug, $permalink ) {
-		$matching_page = get_page_by_path( $permalink );
+		$matching_page = get_page_by_path( $permalink, OBJECT, 'page' );
+
+		/**
+		 * Filters whether to attempt to guess a redirect URL for a 404 request.
+		 *
+		 * Returning a false value from the filter will disable the URL guessing
+		 * and return early without performing a redirect.
+		 *
+		 * @since 5.5.0
+		 *
+		 * @param bool $do_redirect_guess Whether to attempt to guess a redirect URL
+		 *                                for a 404 request. Default true.
+		 */
+		if ( ! $matching_page instanceof WP_Post && apply_filters( 'do_redirect_guess_404_permalink', true ) ) {
```

Comment:
https://developer.wordpress.org/reference/hooks/do_redirect_guess_404_permalink/ shows this function definition:
```php
apply_filters( 'do_redirect_guess_404_permalink', bool $do_redirect_guess )
```
and mentioned that the default value of `$do_redirect_guess` is `true`. Based on that, I wonder if we shouldn't change the check as follows:
```suggestion
		if ( ! $matching_page instanceof WP_Post && apply_filters( 'do_redirect_guess_404_permalink' ) ) {
```

Reply:
We can't, the url you posted does not apply to this, we need to look at the apply_filters() [docs](https://developer.wordpress.org/reference/functions/apply_filters/) where it states the $value is a required arg. 
If we don't supply that we get a critical error.

Reply:
Oh, my bad. You're right, Paulo.